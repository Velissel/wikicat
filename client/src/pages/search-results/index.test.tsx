import React from 'react';
import { RouterProvider, createMemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import SearchResults from './';

jest.mock('../../utils/api');

describe('SearchResults page', () => {
  it('should render', () => {
    const testRouter = createMemoryRouter(
      [
        {
          path: '/',
          element: <SearchResults />,
        },
      ],
      {
        initialEntries: ['/'],
        initialIndex: 0,
      },
    );
    const { container } = render(
      <RouterProvider router={testRouter}></RouterProvider>,
    );

    expect(screen.getByText('Search')).toBeInTheDocument();
    expect(
      screen.getByText('Start search cat breeds by putting keyworkds in'),
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
