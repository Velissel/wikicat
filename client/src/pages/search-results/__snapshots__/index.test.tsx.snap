// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults page should render 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      Cat Search Results
    </h1>
    <form
      action="#"
    >
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="searchTerm"
        >
          Search cat by breed
        </label>
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="searchTerm"
            name="searchTerm"
            type="text"
            value=""
          />
          <button
            class="btn btn-primary"
            type="submit"
          >
            Search
          </button>
        </div>
      </div>
    </form>
    Start search cat breeds by putting keyworkds in
  </div>
</div>
`;
